### DISCUSSION TITLE
Option to redirect the sign-in page to Azure AD / Azure AD B2C sign-in page

### DISCUSSION THREAD

**Discussion by phfrc** [Azure/api-management-developer-portal#2584] [Feature request]
Hello! 😄 

Currently working on a self-hosted version of the APIM Developer Portal, I'm trying to display only what is needed for my customers, hence, only the possibility to sign-in with Azure AD B2C (the only one used in our context).

As you already know, there's a default page "/signin" in the portal that displays two inputs (email, password) and buttons (forget password, Sign in with Azure AD B2C). What is the best way to directly redirect to Azure AD B2C when clicking on "Sign in" from the menu for example? I would also appreciate if you can point me to the right path as well for the sign-out and sign-up.

I followed this [documentation](https://docs.microsoft.com/en-us/azure/api-management/api-management-howto-aad-b2c) for the configuration of Azure AD B2C with custom flows.

Please don't hesitate if you need more informations. Thanking you in advance.

**Comment by ghost** [comment]
@phfrc, thank you for opening this issue. We will triage it within the next few business days.

**Comment by mikebudzynski** [comment]
@phfrc, you can remove the "Sign in" button from the menu and add the "OAuth buttons" widget next to the menu (for example, at the top of the screen / in your Layout).

**Comment by phfrc** [comment]
Thank you for your quick response @mikebudzynski! But what I really need to do is redirect directly to Azure ADB2C sign-in page when landing on the "/signin" in the developer portal. How would you handle this?

Thank you!

**Comment by mikebudzynski** [comment]
@phfrc, this scenario isn't supported at the moment; I will reopen the Issue and mark it as a feature request. The best workaround is to place the OAuth buttons directly in the menu.

**Comment by ghost** [comment]
@phfrc, thank you for requesting this feature.

**Comment by phfrc** [comment]
Hey @mikebudzynski!
Do you think this is something that you guys can implement quickly? What's your process for feature requests?
I'm asking because I was thinking of creating a widget that redirects to the given page.
I'd be happy to talk more about this with you.

**Comment by mikebudzynski** [comment]
> Hey @mikebudzynski!
> Do you think this is something that you guys can implement quickly? What's your process for feature requests?
> I'm asking because I was thinking of creating a widget that redirects to the given page.
> I'd be happy to talk more about this with you.

@phfrc, this improvement is not on our immediate roadmap - we are not able to prioritize it or give you a time estimate at this moment. We have logged this feature here as a feature request; if your company works with Microsoft representatives (account executives or cloud solution architects), you can inform them about this requirement as well.
